lvsfunc/deblock.py

```diff
@@ -185,26 +190,47 @@ def vsdpir(clip: vs.VideoNode, strength: SupportsFloat | vs.VideoNode | None = 2
     clip_32 = depth(clip, 32, dither_type=Dither.ERROR_DIFFUSION)
 
     match mode.lower():
-        case 'deblock': model = DPIRModel.drunet_deblocking_color if not is_gray \
-            else DPIRModel.drunet_deblocking_grayscale
+        case 'deblock': model = DPIRModel.drunet_deblocking_grayscale if is_gray else DPIRModel.drunet_deblocking_color
         case 'denoise': model = DPIRModel.drunet_color if not is_gray else DPIRModel.drunet_gray
         case _: raise TypeError(f"vsdpir: '\"{mode}\" is not a valid mode!'")
 
-    dpir_args |= dict(strength=strength, tiles=tiles, model=model)
+    dpir_args |= dict(tiles=tiles, model=model)
 
     if "backend" not in dpir_args:
-        dpir_args |= dict(backend=Backend.ORT_CUDA if cuda else Backend.OV_CPU)
+        dpir_args |= dict(backend=(Backend.TRT if cuda == 'trt' else Backend.ORT_CUDA) if cuda else Backend.OV_CPU)
 
-    if is_rgb or is_gray:
-        return depth(DPIR(clip_32.std.Limiter(), **dpir_args), bit_depth)
+    def _get_strength_clip(strength: SupportsFloat) -> vs.VideoNode:
+        return clip.std.BlankClip(format=vs.GRAYS, color=float(strength))
+
+    if isinstance(strength, vs.VideoNode):
+        assert strength.format
+
+        if strength.format.color_family != vs.GRAY:
+            raise ValueError("vsdpir: '`strength` must be a GRAY clip!'")
+
+        if strength.format.id != vs.GRAYS:
+            strength = strength.std.Expr("x", vs.GRAYS)
+
+        if strength.width != clip.width or strength.height != clip.height:
+            strength = kernel.scale(strength, clip.width, clip.height)
```

Comment:
Wouldn't it make more sense to error in this situation? This may be unintended behaviour for the end user, unless you believe it will be common for them to pass a clip that doesn't have the same dimensions.